vars/cloudFoundryDeploy.groovy

```diff
@@ -0,0 +1,161 @@
+
+def call(Map parameters = [:]) {
+
+    handlePipelineStepErrors (stepName: STEP_NAME, stepParameters: parameters) {
+
+        def utils = parameters.juStabUtils
+        if (utils == null) {
+            utils = new Utils()
+        }
+
+        def script = parameters.script
+        if (script == null)
+            script = [commonPipelineEnvironment: commonPipelineEnvironment]
+
+        Map config = ConfigurationHelper
+            .loadStepDefaults(this)
+            .mixinGeneralConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS, this, CONFIG_KEY_COMPATIBILITY)
+            .mixinStepConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS, this, CONFIG_KEY_COMPATIBILITY)
+            .mixinStageConfig(script.commonPipelineEnvironment, parameters.stageName?:env.STAGE_NAME, STEP_CONFIG_KEYS, this, CONFIG_KEY_COMPATIBILITY)
+            .mixin(parameters, PARAMETER_KEYS, this, CONFIG_KEY_COMPATIBILITY)
+            .dependingOn('deployTool').mixin('dockerImage')
+            .dependingOn('deployTool').mixin('dockerWorkspace')
+            .withMandatoryProperty('cloudFoundry/org')
+            .withMandatoryProperty('cloudFoundry/space')
+            .use()
+
+        echo "[${STEP_NAME}] General parameters: deployTool=${config.deployTool}, deployType=${config.deployType}, cfApiEndpoint=${config.cloudFoundry.apiEndpoint}, cfOrg=${config.cloudFoundry.org}, cfSpace=${config.cloudFoundry.space}, cfCredentialsId=${config.cloudFoundry.credentialsId}, deployUser=${config.deployUser}"
+
+        utils.unstash 'deployDescriptor'
+
+        if (config.deployTool == 'mtaDeployPlugin') {
+            // set default mtar path
+            config = new ConfigurationHelper(config)
+                .addIfEmpty('mtaPath', config.mtaPath?:findMtar())
+                .use()
+
+            dockerExecute(dockerImage: config.dockerImage, dockerWorkspace: config.dockerWorkspace, stashContent: config.stashContent) {
+                deployMta(config)
+            }
+            return
+        }
+
+        if (config.deployTool == 'cf_native') {
+            def smokeTest = ''
+
+            if (config.smokeTestScript == 'blueGreenCheck.sh') {
+                writeFile file: config.smokeTestScript, text: libraryResource(config.smokeTestScript)
+            } else {
+                utils.unstash 'pipelineConfigAndTests'
+            }
+            smokeTest = '--smoke-test $(pwd)/' + config.smokeTestScript
+            sh "chmod +x ${config.smokeTestScript}"
+
+            echo "[${STEP_NAME}] CF native deployment (${config.deployType}) with cfAppName=${config.cloudFoundry.appName}, cfManifest=${config.cloudFoundry.manifest}, smokeTestScript=${config.smokeTestScript}"
+
+            dockerExecute (dockerImage: config.dockerImage, dockerWorkspace: config.dockerWorkspace, stashContent: config.stashContent, dockerEnvVars: [CF_HOME:"${config.dockerWorkspace}", CF_PLUGIN_HOME:"${config.dockerWorkspace}", STATUS_CODE: "${config.smokeTestStatusCode}"]) {
```

Comment:
Minor: Please add line breaks to increase readability.

Reply:
changed

Reply:
Also, I'd appreciate applying the intellij formatter, as formatting is not consistent regarding spaces etc

Reply:
> intellij formatter

Not everybody is using itellij. IMO it is a fight against windmills to enforce a certain development toolset.